frame/support/src/traits.rs

```diff
@@ -1613,6 +1613,19 @@ pub trait UnfilteredDispatchable {
 	fn dispatch_bypass_filter(self, origin: Self::Origin) -> crate::dispatch::DispatchResultWithPostInfo;
 }
 
+/// Basic origin construction and AccountId type.
+pub trait SystemOrigin {
+	/// The type of account identifiers used by the system.
+	type AccountId;
+
+	/// Create with system none origin and `frame-system::Trait::BaseCallFilter`.
+	fn none() -> Self;
+	/// Create with system root origin and no filter.
+	fn root() -> Self;
+	/// Create with system signed origin and `frame-system::Trait::BaseCallFilter`.
+	fn signed(by: Self::AccountId) -> Self;
+}
```

Comment:
Somehow this should be connected to this right?

https://github.com/paritytech/substrate/blob/master/frame/system/src/lib.rs#L320

Reply:
Yup. I imagine that this connecting link is the macro that impls `SystemOrigin` above